pkg/workspace/controllers/workspace_controller.go

```diff
@@ -826,6 +841,67 @@ func updateWorkspaceWithRetry(ctx context.Context, c client.Client, wObj *kaitov
 	})
 }
 
+// ensureGatewayAPIInferenceExtension reconciles Gateway API Inference Extension components for a Workspace.
+//
+// How it works:
+// 1) Dry-runs preset inference generation to determine if the target workload is a StatefulSet.
+// 2) Renders a Flux OCIRepository and a HelmRelease for the InferencePool chart.
+// 3) Creates the resources if absent; updates them if the desired spec differs.
+// 4) Waits for resources to become ready using the model's inference readiness timeout.
+// 5) Aggregates and returns any errors.
+//
+// Idempotent and safe to call on every reconcile; no-op if preconditions are not met.
+func (c *WorkspaceReconciler) ensureGatewayAPIInferenceExtension(ctx context.Context, wObj *kaitov1beta1.Workspace) error {
```

Comment:
Question:  I notice that the chart include the epp deployment. I assume the helmrelease cr status readiness does not depends on the readiness of this deployment. If so, you need to call checkresourcestatus for this specific deployment at the end of this call. 

A more general question, how should we deal with the case that this deployment is never ready (due to code bug etc, note that we don't own this code), should we always mark this workspace not ready?

Reply:
>  assume the helmrelease cr status readiness does not depends on the readiness of this deployment

Flux HelmRelease's readiness actually does depend on the resources it installs via Helm. For example, I specified a non-existing image in the HelmRelease and it would eventually timeout and the controller would mark HelmRelease as not ready.

<img width="1662" height="79" alt="image" src="https://github.com/user-attachments/assets/550a1d4d-7794-49b9-b3a5-064fd38df486" />

> should we always mark this workspace not ready?

Since the HelmRelease is owned by Workspace, I'd say it should be treated like any other resources owned by Workspace. We should only mark Workspace as Ready when all the child resources are ready.